File: Day3/DeleteAndEarn.py
from hashlib import new
import math
from typing import Counter


class Solutin(object):
    def deletAndEarn(self,nums):
        count=[0] * (max(nums) + 1)
        for num in nums:
            count[num]+=num
        next=0
        prev=0
        cur=0
        for i in range(len(count)):
             next=max(prev+count[i],cur)
             prev=cur
             cur=next
        return next        

File: Day3/DeleteAndEarn.java
package Day3;



public class DeleteAndEarn {
    public int deleteAndEarn(int[] nums) {
       int[] count=new int [10001];
       for (int i : nums) {
        count[i]+=i;
       }
       int prev=0;
       int cur=0,next=0;
       for (int i = 0; i < count.length; i++) {
        next=Math.max(prev+count[i], cur);
        prev=cur;
        cur=next;
       }


        return next;
    }
}
